### PR TITLE
feat: require payment plans to provision spaces

### PR DESCRIPTION
**Pull request by travis**
Most of the heavy lifting here is done by upgrading the `@web3-storage/upload-api` dependency. There is a new "feature flag" in the context that changes whether payment plans are required for provisioning - I'm going to set it to `false` for now in production and we can flip it once we're ready to turn this on.

I spent a bit of time working toward extending our integration tests to exercise the Stripe webhook handler, but it turned into a big enough project that I decided to defer and recorded what I learned in an issue here:

https://github.com/web3-storage/w3infra/issues/267

### CODE REVIEW

File: .env.tpl [hunk 1]
```diff
@@ -38,3 +38,5 @@ SATNAV_BUCKET_NAME = ''
 MAILSLURP_API_KEY = ''
 MAILSLURP_TIMEOUT = '120000'
 
+# Feature flags
+REQUIRE_PAYMENT_PLAN = 'true'
```
Comment (vasco-santos) [inline .env.tpl:42]: would be good to test this in staging. I am not 100% sure if swapping env var is enough, or we need to re-deploy. So knowing that can help us know what we need to do

**Review by vasco-santos — APPROVED** [review]
changes look good to me. We just need to see what is going on in inetgration tests by upgrading the `upload-api`

File: upload-api/test/helpers/ucan.js [hunk 2]
```diff
@@ -295,6 +295,7 @@ export async function executionContextToUcantoTestServerContext (t) {
 
   return {
     ...serviceContext,
+    grantAccess: async () => {},
```
Comment (alanshaw) [inline upload-api/test/helpers/ucan.js]:
```suggestion
```

**Review by alanshaw — APPROVED** [review]

File: upload-api/test/helpers/ucan.js (hunk 2 above)
Comment (travis) [reply upload-api/test/helpers/ucan.js]: oh lol we were looking at this at precisely the same time - fixed, thx!

**Pull request merged**